src/Symfony/Component/Translation/Util/IntlMessageConverter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Util;
+
+/**
+ * Convert from Symfony 3's plural syntax to Intl message format.
+ * {@link https://messageformat.github.io/messageformat/page-guide}.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class IntlMessageConverter
+{
+    public static function convert(string $message): string
+    {
+        $array = self::getMessageArray($message);
+
+        if (1 === \count($array) && isset($array[0])) {
+            return preg_replace('|%(.*?)%|s', '{$1}', $message);
+        }
+
+        $icu = self::buildIcuString($array);
+
+        return $icu;
+    }
+
+    /**
+     * Get an ICU like array.
+     */
+    private static function getMessageArray(string $message): array
+    {
+        $parts = array();
+        if (preg_match('/^\|++$/', $message)) {
+            $parts = explode('|', $message);
+        } elseif (preg_match_all('/(?:\|\||[^\|])++/', $message, $matches)) {
+            $parts = $matches[0];
+        }
+
+        $intervalRegexp = <<<'EOF'
+/^(?P<interval>
+    ({\s*
+        (\-?\d+(\.\d+)?[\s*,\s*\-?\d+(\.\d+)?]*)
+    \s*})
+
+        |
+
+    (?P<left_delimiter>[\[\]])
+        \s*
+        (?P<left>-Inf|\-?\d+(\.\d+)?)
+        \s*,\s*
+        (?P<right>\+?Inf|\-?\d+(\.\d+)?)
+        \s*
+    (?P<right_delimiter>[\[\]])
+)\s*(?P<message>.*?)$/xs
+EOF;
+
+        $standardRules = array();
+        foreach ($parts as $part) {
```

Comment:
What if `$parts` is empty ? I would have add a check not to continue if `empty($parts)` is true